crates/wasmtime/src/compile.rs

```diff
@@ -110,9 +116,10 @@ pub(crate) fn build_artifacts<T: FinishedObject>(
 /// The output artifact here is the serialized object file contained within
 /// an owned mmap along with metadata about the compilation itself.
 #[cfg(feature = "component-model")]
-pub(crate) fn build_component_artifacts<T: FinishedObject>(
+pub(crate) fn build_component_artifacts<'a, T: FinishedObject>(
     engine: &Engine,
     binary: &[u8],
+    dwarf_package_binary: Option<&'a [u8]>,
```

Comment:
This is probably best left as unconditionally `None` for components since components will need more support for loading multiple possible `*.dwp` files or something like that in the future.

Reply:
Yes, I'd like to tackle that after this.